src/main/java/yevhen/bookstore/controller/BookController.java

```diff
@@ -24,48 +26,53 @@
 @Tag(name = "Online book store managing", description = "Endpoints for managing book store")
 @RequiredArgsConstructor
 @RestController
-@RequestMapping("/books")
+@RequestMapping("/api/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all available books")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
     @Operation(summary = "Get one book by id", description = "Get one book by id")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
     @Operation(summary = "Create a new book", description = "Create a new book")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
     @Operation(summary = "Update a book by id", description = "Update a book with certain id")
-    public BookDto updateBookById(@RequestBody CreateBookRequestDto requestDto,
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public BookDto updateBookById(@RequestBody @Valid CreateBookRequestDto requestDto,
                                   @PathVariable Long id) {
         return bookService.updateBookById(requestDto, id);
     }
 
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
     @Operation(summary = "Delete one book by id", description = "Delete one book by id")
-    public void deleteBookById(@PathVariable Long id) {
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public void deleteBookById(Authentication authentication, @PathVariable Long id) {
```

Comment:
```suggestion
    public void deleteBookById(Authentication authentication, @PathVariable @Positive Long id) {
```

Check all path variables and use `@Validated` on class level